recipes/up-cpp/all/conanfile.py

```diff
@@ -0,0 +1,159 @@
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_testing": [True, False],
+        "build_unbundled": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": False,
+        "build_testing": False,
+        "build_unbundled": False,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    # unclear on what I should choose for msvc and Visual Studio
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "13",
+            "clang": "13",
+            "gcc": "11",
+            "msvc": "191",
+            "Visual Studio": "15",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+        self.folders.build = os.path.join("build", str(self.settings.build_type))
+        self.folders.install = "install"
+
+    def build_requirements(self):
+        self.build_requires("cmake/[>=3.18.0]")
+
+    def requirements(self):
+        self.requires("protobuf/3.21.12")
+        self.requires("spdlog/1.13.0")
+        if self.options.build_testing:
+            self.requires("gtest/1.14.0")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+        if is_msvc(self):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} need to have paths in CMake files updated to handle both \ and /."
+            )
+        if self.settings.compiler == "clang" and stdcpp_library(self) == "c++":
+            raise ConanInvalidConfiguration(
+                f"{self.ref} Experiencing some interference between math.h and the protobuf generated udiscovery.pb.h"
+                  "when compiling with libc++ for some reason."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+        # we clone protos from the appropriate repo at the appropriate tag to our local root to build
+        rmdir(self, self.proto_containing_folder)
+
+        git = Git(self)
+        git_method = os.getenv("GIT_METHOD", "https")
+        git.clone(self.conan_data["proto"][self.version][git_method], target=self.proto_containing_folder)
+        git.folder=self.proto_containing_folder
+        git.checkout(self.conan_data["proto"][self.version]["tag"])
```

Comment:
```suggestion
        git = Git(self)
        git_method = os.getenv("GIT_METHOD", "https")
        git.clone(self.conan_data["proto"][self.version][git_method], target=self.proto_containing_folder)
        git.folder=self.proto_containing_folder
        git.checkout(self.conan_data["proto"][self.version]["tag"])
```
Need more detail about it, we don't use Git in ConanCenterIndex, instead, we download the tar.gz to be sure about the immutability of the package.